OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -157,6 +172,17 @@ static IEnumerable<CPos> GetMinefieldCells(CPos start, CPos end, WDist depth)
 						yield return new CPos(i, j);
 		}
 
+		bool CanLayMineOnCell(Actor self, CPos cell)
+		{
+			var terrainType = self.World.Map.GetTerrainInfo(cell).Type;
```

Comment:
Doing terrain info lookups even when not needed (which is the default case) isn't great.
I suggest writing this as
```
if (Info.TerrainTypes.Count == 0)
	return true;

var terrainType = self.World.Map.GetTerrainInfo(cell).Type;
return Info.TerrainTypes.Contains(terrainType);
```

Reply:
Good point. Fixed.